demos/rag_agentic/notebooks/level4_agentic_and_mcp.ipynb

```diff
@@ -0,0 +1,129 @@
+{
+  "metadata": {
+    "kernelspec": {
+      "name": "python",
+      "display_name": "Python (Pyodide)",
+      "language": "python"
+    },
+    "language_info": {
+      "codemirror_mode": {
+        "name": "ipython",
+        "version": 3
+      },
+      "file_extension": ".py",
+      "mimetype": "text/x-python",
+      "name": "python",
+      "nbconvert_exporter": "python",
+      "pygments_lexer": "ipython3",
+      "version": "3.11.7"
+    }
+  },
+  "nbformat_minor": 5,
+  "nbformat": 4,
+  "cells": [
+    {
+      "id": "b5ecd807-fb1c-41eb-a948-365e57396d90",
+      "cell_type": "markdown",
+      "source": "# Level 4: Agentic & MCP (Medium Difficulty)\n\nThis tutorial is aimed at those already familiar with basic Agentic workflows. It is meant to showcase  **sequential tool calls** or **conditional logic** within the context of an agentic workflow.\n\n## Overview\n\nIn this tutorial we will be connecting to a llamastack instance, building an agent with various tools available to it, and inferencing against the agent.\n\n## Prerequisites\n\nBefore starting, ensure you have the following:\n- Access to an openshift cluster.\n- A deployment of the [openshift MCP server](https://github.com/opendatahub-io/llama-stack-on-ocp/tree/main/mcp-servers/openshift) within the openshift cluster (see the [deployment manifests](https://github.com/opendatahub-io/llama-stack-on-ocp/tree/main/kubernetes/mcp-servers/openshift-mcp) for assistance with this).\n- User variables configured (e.g., inference_model, TAVILY_SEARCH_API_KEY, LLAMA_STACK_ENDPOINT).\n- A Tavily API key is required. You can register for one at https://tavily.com/.\n\n## General Setup - Agnostic to all Queries\n\nThese steps will be the same for all 3 queries, with the exception of inputing the Tavily API key - this will only be used for query 2, and so could be ommitted if one only wants to run demos 1 or 3.\n\n### Configuring logging",
```

Comment:
I'm not sure you need to make "access to an openShift Cluster" a pre-req here since below you make it clear you can use a local or remote instance. Probably just want to make clear that a llama stack server is required. And you can pull the links to "remote setup guide" and "local setup guide" from the section below to here.